apache2/msc_xml.c

```diff
@@ -86,18 +215,52 @@ int xml_process_chunk(modsec_rec *msr, const char *buf, unsigned int size, char
 
         */
 
-        msr->xml->parsing_ctx = xmlCreatePushParserCtxt(NULL, NULL, buf, size, "body.xml");
-        if (msr->xml->parsing_ctx == NULL) {
-            *error_msg = apr_psprintf(msr->mp, "XML: Failed to create parsing context.");
-            return -1;
+        if (msr->txcfg->parse_xml_into_args != MSC_XML_ARGS_ONLYARGS) {
```

Comment:
This will also apply to `MSC_XML_ARGS_OFF`. Is that correct?

Reply:
Yes, exactly. That's how it must behave.

If the user set the `SecParseXMLintoArgs` to `Off` (which is the default), then we definitely **need** this context.
Also, if the user set the directive to `On`, then it means user wants to parse XML nodes into ARGS **plus** he wants to see them under `XML` target too - therefore we definitely need this context too.

If the user set the directive to `OnlyArg`, then we **don't need** this context.